databricks/koalas/series.py

```diff
@@ -3409,6 +3409,80 @@ def replace(self, to_replace=None, value=None, regex=False) -> 'Series':
 
         return self._with_new_scol(current)
 
+    def where(self, cond, other=np.nan):
+        """
+        Replace values where the condition is False.
+
+        Parameters
+        ----------
+        cond : boolean Series
+            Where cond is True, keep the original value. Where False,
+            replace with corresponding value from other.
+        other : scalar, Series
+            Entries where cond is False are replaced with corresponding value from other.
+
+        Returns
+        -------
+        Series
+
+        Examples
+        --------
+
+        >>> s1 = ks.Series([0, 1, 2, 3, 4])
+        >>> s2 = ks.Series([100, 200, 300, 400, 500])
+        >>> s1.where(s1 > 0)
+        0    NaN
+        1    1.0
+        2    2.0
+        3    3.0
+        4    4.0
+        Name: 0, dtype: float64
+
+        >>> s1.where(s1 > 1, 10)
+        0    10
+        1    10
+        2     2
+        3     3
+        4     4
+        Name: 0, dtype: int64
+
+        >>> s1.where(s1 > 1, s1 + 100)
+        0    100
+        1    101
+        2      2
+        3      3
+        4      4
+        Name: 0, dtype: int64
+
+        >>> s1.where(s1 > 1, s2)
+        0    100
+        1    200
+        2      2
+        3      3
+        4      4
+        Name: 0, dtype: int64
+        """
+        current = F.when(cond._internal.data_scols[0], self._scol)
+        if isinstance(other, Series):
+            self_sdf = self._internal.sdf
+            other_sdf = other._internal.sdf
+            if self_sdf == other_sdf:
+                current = current.otherwise(other._scol)
+                result = self._with_new_scol(current)
+            else:
+                temp_col = self.name + "_"
+                temp_idx = self._index_map[0][0]
+                other_temp = other_sdf.withColumn(temp_col, other_sdf[other.name])
+                new_sdf = self_sdf.join(other_temp, temp_idx).sort(temp_idx)
```

Comment:
Yea, I think you can do, for instance,

```python
self_sdf[temp_col] = ...
current.otherwise(other_temp[temp_col])
```

Keep it same as https://github.com/databricks/koalas/commit/8367c93343da181fe1a3b0702efe75b97f522460#diff-35732ddb08a2af7a8c8a0c3c4ad54b52R514 

Adding temp columns isn't a good idea. We might have to use `utils.align_diff_series` there as well.

Reply:
Thanks for commenting!! i'm trying to use `align_diff_series`, and have got a little confused since there is no usecase examples of `align_diff_series` 😭